metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/TransformationService.java

```diff
@@ -0,0 +1,111 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.rest.service;
+
+import org.apache.metron.common.dsl.Context;
+import org.apache.metron.common.dsl.ParseException;
+import org.apache.metron.common.dsl.StellarFunctionInfo;
+import org.apache.metron.common.dsl.functions.resolver.SingletonFunctionResolver;
+import org.apache.metron.common.field.transformation.FieldTransformations;
+import org.apache.metron.common.stellar.StellarProcessor;
+import org.apache.metron.rest.model.StellarFunctionDescription;
+import org.apache.metron.rest.model.TransformationValidation;
+import org.json.simple.JSONObject;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class TransformationService {
+
+    public Map<String, Boolean> validateRules(List<String> rules) {
+        Map<String, Boolean> results = new HashMap<>();
+        StellarProcessor stellarProcessor = new StellarProcessor();
+        for(String rule: rules) {
+            try {
+                boolean result = stellarProcessor.validate(rule, Context.EMPTY_CONTEXT());
+                results.put(rule, result);
+            } catch (ParseException e) {
+                results.put(rule, false);
+            }
+        }
+        return results;
+    }
+
+    public Map<String, Object> validateTransformation(TransformationValidation transformationValidation) {
+        JSONObject sampleJson = new JSONObject(transformationValidation.getSampleData());
+        transformationValidation.getSensorParserConfig().getFieldTransformations().forEach(fieldTransformer -> {
+                    fieldTransformer.transformAndUpdate(sampleJson, transformationValidation.getSensorParserConfig().getParserConfig(), Context.EMPTY_CONTEXT());
+                }
+        );
+        return sampleJson;
+    }
+
+    public FieldTransformations[] getTransformations() {
+        return FieldTransformations.values();
+    }
+
+    public List<StellarFunctionDescription> getStellarFunctions() {
+        List<StellarFunctionDescription> stellarFunctionDescriptions = new ArrayList<>();
+        Iterable<StellarFunctionInfo> stellarFunctionsInfo = SingletonFunctionResolver.getInstance().getFunctionInfo();
+        stellarFunctionsInfo.forEach(stellarFunctionInfo -> {
+            stellarFunctionDescriptions.add(new StellarFunctionDescription(
+                    stellarFunctionInfo.getName(),
+                    stellarFunctionInfo.getDescription(),
+                    stellarFunctionInfo.getParams(),
+                    stellarFunctionInfo.getReturns()));
+        });
+        return stellarFunctionDescriptions;
+    }
+
+    private List<String> simpleFunctionNames = Arrays.asList(
```

Comment:
Can you pull these from the classpath based on the Stellar annotation?  I presume "simple" in this case are stellar functions with one argument (that might be a relevant comment in the code actually).  You should be able to get the list of stellar functions via reflection and then get the number of params to filter them out.